core/src/main/java/org/apache/spark/shuffle/sort/io/DefaultShufflePartitionWriter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort.io;
+
+import org.apache.spark.api.shuffle.ShufflePartitionWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+
+public class DefaultShufflePartitionWriter implements ShufflePartitionWriter {
```

Comment:
This is rare, but in this case I think it makes more sense for this to be a static inner class of `DefaultMapOutputWriter` - the interactions between the length stored here and usage in `DefaultMapOutputWriter#commitAllPartitions` would be clearer this way. But I'd like a second opinion from @yifeih.

Reply:
Yeah, I am okay with either. I thought this was more isolated.

Reply:
I don't have a strong preference either way, although I did find myself needed to go to the other class to understand this one, so placing it together could help.